src/clr-angular/forms/datepicker/date-input.spec.ts

```diff
@@ -0,0 +1,288 @@
+            });
+
+            describe("Typescript API", () => {
+                it("gets the placeholder from the IO service", () => {
+                    // since we are testing with en-US
+                    expect(context.clarityDirective.placeholderText).toBe("MM/DD/YYYY");
+                });
+
+                it("gets whether the datepicker is enabled or not", () => {
+                    expect(enabledService.isEnabled).toBe(true);
+                    expect(context.clarityDirective.inputType).toBe("text");
+
+                    enabledService.fakeIsEnabled = false;
+                    expect(enabledService.isEnabled).toBe(false);
+
+                    context.detectChanges();
+
+                    expect(context.clarityDirective.inputType).toBe("date");
+                });
+
+                it("sets the inputDate if the value of the input is valid", () => {
+                    const testEl: HTMLInputElement = <HTMLInputElement>document.createElement("INPUT");
+                    testEl.value = "01/02/2015";
+
+                    expect(dateIOService.inputDate).toBeUndefined();
+
+                    context.clarityDirective.onValueChange(testEl);
+
+                    expect(dateIOService.inputDate).not.toBeUndefined();
+                    expect(dateIOService.inputDate).toBe("01/02/2015");
+                });
+
+                it("sets the inputDate to an empty string if the value of the input is invalid", () => {
+                    const testEl: HTMLInputElement = <HTMLInputElement>document.createElement("INPUT");
+                    testEl.value = "01/02/201";
+
+                    expect(dateIOService.inputDate).toBeUndefined();
+
+                    context.clarityDirective.onValueChange(testEl);
+
+                    expect(dateIOService.inputDate).not.toBeUndefined();
+                    expect(dateIOService.inputDate).toBe("");
+                });
+
+                it("sets the date on the IOService", () => {
+                    const date: Date = new Date();
+
+                    expect(dateIOService.date).toBeUndefined();
+
+                    context.clarityDirective.date = date;
+
+                    expect(dateIOService.date).not.toBeUndefined();
+                    expect(dateIOService.date.getDate()).toBe(date.getDate());
+                    expect(dateIOService.date.getMonth()).toBe(date.getMonth());
+                    expect(dateIOService.date.getFullYear()).toBe(date.getFullYear());
+
+                    const inValidDate: Date = new Date("Test");
+
+                    context.clarityDirective.date = inValidDate;
+                    expect(dateIOService.date).toBeNull();
+                });
+            });
+
+            describe("Host Bindings & Listeners", () => {
+                it("binds the placeholderText to the placeholder of the input", () => {
+                    expect(context.clarityElement.placeholder).toBe("MM/DD/YYYY");
```

Comment:
Should test that we do not override a custom placeholder.

Reply:
fixed